### PR TITLE
Release 2.0

### PR DESCRIPTION
**Pull request by Tydesson**
- Dropped support for Symfony 2
- Added Unit Testing - Issue #13 
- Removed checkPendingAction() method from PendingActionService

A big hug to @Pierstoval for his help on tricky unit tests.

### CODE REVIEW

File: Tests/PhpunitTest.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Class PhpunitTest
+ * @package ClaviculaNox\PendingActionsBundle\Tests
+ */
+class PhpunitTest extends \PHPUnit\Framework\TestCase
+{
+    public function testPhpunit()
+    {
+        $this->assertTrue(true);
```
Comment (Pierstoval) [inline Tests/PhpunitTest.php:22]: Such Volkswagen style 🚗

File: Tests/app/config/config_test.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    trusted_hosts: ~
+    secret: "test"
+    translator:      { fallback: "fr" }
+    test: ~
+    templating:
+        engines: ['twig']
+    router:
+        resource: "%kernel.root_dir%/config/routing.yml"
+
+
```
Comment (Pierstoval) [inline Tests/app/config/config_test.yml]: Extra newline not needed here

File: Tests/app/config/config_test.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+framework:
+    trusted_hosts: ~
+    secret: "test"
+    translator:      { fallback: "fr" }
```
Comment (Pierstoval) [inline Tests/app/config/config_test.yml]: Extra spaces useless here

File: Tests/app/console [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env php
```
Comment (Pierstoval) [inline Tests/app/console]: Do you really need a console for test in this bundle?

File: Tests/autoload.php.dist [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?php
```
Comment (Pierstoval) [inline Tests/autoload.php.dist]: No need to name it `.dist`, you're not gonna change this, and you shouldn't

File: Tests/autoload.php.dist [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    'Doctrine'               => $vendorDir.'/doctrine/lib',
+    'DoctrineExtensions'     => $vendorDir.'/beberlei-doctrine-extensions/lib',
+));
+$loader->registerPrefixes(array(
+    'Twig_' => $vendorDir.'/twig/lib',
+));
+$loader->register();
+
+spl_autoload_register(function($class) {
+    if (0 === strpos($class, 'Kitpages\\DataGridBundle\\')) {
```
Comment (Pierstoval) [inline Tests/autoload.php.dist]: I see this bundle is not even required in `composer.json`

File: composer.json [hunk 7]
```diff
@@ -13,13 +13,28 @@
     ],
 
     "require": {
-        "php":                      ">=5.5.0",
-        "symfony/framework-bundle": "~2.7|~3.0",
+        "php":                      "^7.0",
+        "symfony/framework-bundle": "~3.0",
         "doctrine/doctrine-bundle": "~1.4",
         "doctrine/orm":             "~2.4"
```
Comment (Pierstoval) [inline composer.json]: Dependencies can use `^` as flag instead of `~`

File: composer.json [hunk 8]
```diff
@@ -13,13 +13,28 @@
...
     },
 
+    "require-dev": {
+        "phpunit/phpunit": "~6.1"
```
Comment (Pierstoval) [inline composer.json]: `^6.0`  sshould be fine here

File: composer.json [hunk 9]
```diff
@@ -13,13 +13,28 @@
...
+    },
+
     "autoload": {
-        "psr-4": { "ClaviculaNox\\PendingActionsBundle\\": "" }
+        "psr-4": {
+            "ClaviculaNox\\PendingActionsBundle\\": ""
+        },
+        "exclude-from-classmap": [
+            "/Tests/"
```
Comment (Pierstoval) [inline composer.json]: No need for a slash at the beginning here, to avoid confusion with global paths

**Review by Pierstoval — CHANGES_REQUESTED** [review]

File: Tests/autoload.php.dist (hunk 6 above)
Comment (Tydesson) [reply Tests/autoload.php.dist]: Bad copy paste :)

File: Tests/ServiceHandlerTest.php [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    public function testRegistration()
+    {
+        $Action = $this->getPendingAction();
+
+        if ($Action instanceof PendingAction) {
+            $result = true;
+        } else {
+            $result = false;
+        }
+        $this->assertTrue($result);
```
Comment (Pierstoval) [inline Tests/ServiceHandlerTest.php]: Could be inlined:

```php
static::assertTrue($action instanceof PendingAction);
```

And the var name should be lowercase, as of [Symfony coding standards](http://symfony.com/doc/current/contributing/code/standards.html#naming-conventions)

File: Tests/ServiceHandlerTest.php [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    public function testGroup()
+    {
+        $Action = $this->getPendingAction();
+
+        $this->assertEquals($this->group, $Action->getActionGroup());
```
Comment (Pierstoval) [inline Tests/ServiceHandlerTest.php]: Same here about the `$Action` var, it should be named `$action` instead

File: Tests/App/config/config_test.yaml [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+framework:
+    trusted_hosts: ~
+    secret: "test"
+    test: ~
+
+doctrine:
+    dbal:
+        driver:   pdo_sqlite
+        dbname:   pendingaction_bundle_test
+        user:     root
```
Comment (Pierstoval) [inline Tests/App/config/config_test.yaml]: `dbname` and `user` are useless when using sqlite

File: Tests/App/config/config_test.yaml [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+        charset:  UTF8
+        path:     '%database_path%'
+    orm:
+        auto_generate_proxy_classes: true
+        auto_mapping: false
+        mappings:
```
Comment (Pierstoval) [inline Tests/App/config/config_test.yaml:17]: This part is useless too: doctrine should automatically register bundle entities if they're in the `Entity` bundle directory

File: Tests/App/console [hunk 14]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env php
```
Comment (Pierstoval) [inline Tests/App/console]: I repeat: Do you really need a console for tests in this bundle?

**Review by Pierstoval — CHANGES_REQUESTED** [review]

File: Tests/App/config/config_test.yaml (hunk 13 above)
Comment (Tydesson) [reply Tests/App/config/config_test.yaml:17]: Doesn't seem to work without this part of the code

File: Tests/App/config/config_test.yaml [hunk 15]
```diff
@@ -21,3 +21,9 @@ doctrine:
                 type: annotation
                 dir: '%kernel.root_dir%/../../Entity'
                 prefix: ClaviculaNox\PendingActionsBundle\Entity
+
+services:
+    fake.service:
+         class: ClaviculaNox\PendingActionsBundle\Tests\FakeBundle\Classes\FakeService
+         arguments:
+             []
```
Comment (Pierstoval) [inline Tests/App/config/config_test.yaml]: If arguments are empty, the whole `arguments:` statement can be removed. And a newline is needed at the end of the file

File: Tests/FakeBundle/Classes/FakeService.php [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+ */
+class FakeService
+{
+    const MODE = "defaultMode";
+    const TITLE = "defaultTitle";
+
+    public function __construct()
+    {
+
+    }
```
Comment (Pierstoval) [inline Tests/FakeBundle/Classes/FakeService.php]: If constructor is empty, it's useless

**Review by Pierstoval — CHANGES_REQUESTED** [review]

File: Tests/FakeBundle/Classes/FakeService.php [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+     * @param string $title
+     * @return bool
+     */
+    public function fakeMethod($mode, $title)
+    {
+        if ($mode == FakeService::MODE && $title == FakeService::TITLE) {
+            return true;
+        }
+
+        return false;
```
Comment (Pierstoval) [inline Tests/FakeBundle/Classes/FakeService.php]: Could also be inlined to `return $mode == FakeService::MODE && $title == FakeService::TITLE;`

File: Tests/ServiceHandlerTest.php [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        $Action = $this->getPendingAction();
+
+        if ($Action instanceof PendingAction) {
+            $result = true;
+        } else {
+            $result = false;
+        }
+
+        $this->assertTrue($result);
```
Comment (Pierstoval) [inline Tests/ServiceHandlerTest.php]: Even if previous review was dismissed, this part could be inlined to `$this->assertTrue($Action instanceof PendingAction);`

**Review by Pierstoval — CHANGES_REQUESTED** [review]

**Pull request merged**